compiler/rustc_trait_selection/src/solve/assembly.rs

```diff
@@ -98,52 +98,75 @@ pub(super) trait GoalKind<'tcx>: TypeFoldable<'tcx> + Copy + Eq {
         assumption: ty::Predicate<'tcx>,
     ) -> QueryResult<'tcx>;
 
+    // A type implements an `auto trait` if its components do as well. These components
+    // are given by built-in rules from [`instantiate_constituent_tys_for_auto_trait`](super::trait_goals::structural_traits::instantiate_constituent_tys_for_auto_trait).
     fn consider_auto_trait_candidate(
         ecx: &mut EvalCtxt<'_, 'tcx>,
         goal: Goal<'tcx, Self>,
     ) -> QueryResult<'tcx>;
 
+    // A trait alias holds if the RHS traits and `where` clauses hold.
     fn consider_trait_alias_candidate(
         ecx: &mut EvalCtxt<'_, 'tcx>,
         goal: Goal<'tcx, Self>,
     ) -> QueryResult<'tcx>;
 
+    // A type is `Copy` or `Clone` if its components are `Sized`. These components
+    // are given by built-in rules from [`instantiate_constituent_tys_for_sized_trait`](super::trait_goals::structural_traits::instantiate_constituent_tys_for_sized_trait).
     fn consider_builtin_sized_candidate(
         ecx: &mut EvalCtxt<'_, 'tcx>,
         goal: Goal<'tcx, Self>,
     ) -> QueryResult<'tcx>;
 
+    // A type is `Copy` or `Clone` if its components are `Copy` or `Clone`. These
+    // components are given by built-in rules from [`instantiate_constituent_tys_for_copy_clone_trait`](super::trait_goals::structural_traits::instantiate_constituent_tys_for_copy_clone_trait).
```

Comment:
unhappy with the line length here :grin:

I guess "[`instantiate_constituent_tys_for_copy_clone_trait`]" doesn't work and writing "[super::trait_goals::structural_traits::instantiate_constituent_tys_for_copy_clone_trait]" is ugly.

Funny idea: import `structural_traits::*` with a `#[cfg(doc)]` and then "[`instantiate_constituent_tys_for_copy_clone_trait`]" should work I think

Reply:
Hmmmmm good idea. Yeah, agree about the line length.